### PR TITLE
Allow php-http/httplug 2.x

### PR DESCRIPTION
**Pull request by odolbeau**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -15,15 +15,15 @@
...
         "php-http/multipart-stream-builder": "^1.0",
-        "php-http/client-common": "^1.1",
+        "php-http/client-common": "^1.1 | ^2.0",
         "php-http/discovery": "^1.0"
     },
     "require-dev": {
         "phpunit/phpunit": "^5.7",
         "php-http/message-factory": "^1.0",
-        "php-http/curl-client": "^1.6",
+        "php-http/guzzle6-adapter": "^1.6 | ^2.0",
```
Comment (odolbeau) [inline composer.json]: There isn't any version of `curl-client` compatible with `httplug` 2.x, that's why I used `guzzle6-dapter` here.

File: composer.json [hunk 2]
```diff
@@ -15,15 +15,15 @@
...
         "php-http/multipart-stream-builder": "^1.0",
-        "php-http/client-common": "^1.1",
+        "php-http/client-common": "^1.1 | ^2.0",
         "php-http/discovery": "^1.0"
     },
     "require-dev": {
         "phpunit/phpunit": "^5.7",
         "php-http/message-factory": "^1.0",
-        "php-http/curl-client": "^1.6",
+        "php-http/guzzle6-adapter": "^1.1 | ^2.0",
```
Comment (odolbeau) [inline composer.json]: There isn't any version of `curl-client` compatible with `httplug` 2.x, that's why I used `guzzle6-adapter` here.

File: composer.json [hunk 3]
```diff
@@ -15,15 +15,15 @@
...
         "php-http/multipart-stream-builder": "^1.0",
-        "php-http/client-common": "^1.1",
+        "php-http/client-common": "^1.1 | ^2.0",
         "php-http/discovery": "^1.0"
     },
     "require-dev": {
         "phpunit/phpunit": "^5.7",
         "php-http/message-factory": "^1.0",
-        "php-http/curl-client": "^1.6",
+        "php-http/guzzle6-adapter": "^1.1.1 | ^2.0",
```
Comment (odolbeau) [inline composer.json:26]: There isn't any version of `curl-client` compatible with `httplug` 2.x, that's why I used `guzzle6-adapter` here.
Version 1.1 contains a bug which make tests fail, that's why I used `^1.1.1`.

File: composer.json [hunk 4]
```diff
@@ -15,15 +15,15 @@
     ],
     "require": {
         "php": "^7.0",
-        "php-http/httplug": "^1.0",
+        "php-http/httplug": "^1.0 | ^2.0",
         "php-http/multipart-stream-builder": "^1.0",
-        "php-http/client-common": "^1.1",
+        "php-http/client-common": "^1.1 | ^2.0",
```
Comment (Nyholm) [inline composer.json]: Should be `^1.9 || ^2.0`

File: composer.json (hunk 3 above)
Comment (Nyholm) [reply composer.json:26]: No, Not yet.

Comment (odolbeau) [reply composer.json:26]: What do you mean by "no, not yet"?

Comment (Nyholm) [reply composer.json:26]: Sorry, I read this too fast. 

> Is any version of `curl-client` compatible with `httplug` 2.x?

My bad. :)

I agree with your change.

File: composer.json (hunk 4 above)
Comment (odolbeau) [reply composer.json]: Done

File: composer.json (hunk 3 above)
Comment (odolbeau) [reply composer.json:26]: Great. :)

**Comment by Nyholm** [comment]
Thank you

**Comment by odolbeau** [comment]
There is a strange error with php 7.0 & `--prefer-lowest` option: https://travis-ci.com/odolbeau/localise.biz/jobs/176533613
I don't understand why composer isn't happy with this config (works fine locally on php7.3).

**Comment by Nyholm** [comment]
The issue is that we rely on a package that dont support php7.0. 

I fix this in #28

**Comment by XWB** [comment]
@Nyholm Can you tag a new version with this change?

**Pull request merged**